mono/metadata/mempool.c

```diff
@@ -124,6 +136,18 @@ mono_mempool_new_size (int initial_size)
  *
  * Free all memory associated with this pool.
  */
+#if defined(__has_feature)
+#if __has_feature(thread_sanitizer)
+// clang's ThreadSanitizer detects races of total_bytes_allocated and pool->d.allocated throughout the functions
+//   * mono_mempool_alloc
+//   * mono_mempool_new_size
+//   * mono_mempool_destroy
+// while these races could lead to wrong values, total_bytes_allocated is just used for debugging / reporting and since
+// the mempool.c functions are called quite often, a discussion led the the conclusion of ignoring these races:
+// https://bugzilla.xamarin.com/show_bug.cgi?id=57936
+__attribute__ ((no_sanitize("thread")))
+#endif
+#endif
```

Comment:
Definitely, I just added a fixup commit! Generally, I think the `NO_SANITIZE_THREAD` macro can be immensely useful if we decide to explicitly whitelist more races of that nature ... there are ~ 50 races that are directly connected to debugging / reporting variables. As I am new and don't want to simply put a macro SOMEwhere in the code - where am I supposed to put it? :) Should I create a new header file (e.g. `sanitizing.h` in `mono/utils/`) that can be used for more sanitizing macros in the future as the overall plan is to include more of clang's sanitizers?